pycsghub/repository.py

```diff
@@ -263,6 +264,109 @@ def commits_to_push(self, work_dir: Union[str, Path]) -> int:
         except subprocess.CalledProcessError as exc:
             raise EnvironmentError(exc.stderr)
 
+
+    def track_large_files(self, work_dir: str, pattern: str = ".") -> List[str]:
+        files_to_be_tracked_with_lfs = []
+
+        deleted_files = self.list_deleted_files(work_dir=work_dir)
+
+        for filename in self.list_files_to_be_staged(work_dir=work_dir, pattern=pattern):
+            if filename in deleted_files:
+                continue
+
+            path_to_file = os.path.join(os.getcwd(), work_dir, filename)
+            size_in_mb = os.path.getsize(path_to_file) / (1024 * 1024)
+
+            if size_in_mb >= 1 and not self.is_tracked_with_lfs(filename=path_to_file) and not self.is_git_ignored(filename=path_to_file):
+                self.lfs_track(work_dir=work_dir,patterns=filename)
+                files_to_be_tracked_with_lfs.append(filename)
+
+        self.lfs_untrack(work_dir=work_dir, patterns=deleted_files)
+
+        return files_to_be_tracked_with_lfs
+
+    def list_files_to_be_staged(self, work_dir: str, pattern: str = ".") -> List[str]:
+        try:
+            p = self.run_subprocess("git ls-files --exclude-standard -mo".split() + [pattern], work_dir)
+            if len(p.stdout.strip()):
+                files = p.stdout.strip().split("\n")
+            else:
+                files = []
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        return files
+    
+    def list_deleted_files(self, work_dir: str) -> List[str]:
+        try:
+            git_status = self.run_subprocess("git status -s", work_dir).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        if len(git_status) == 0:
+            return []
+
+        modified_files_statuses = [status.strip() for status in git_status.split("\n")]
+        deleted_files_statuses = [status for status in modified_files_statuses if "D" in status.split()[0]]
+        deleted_files = [status.split()[-1].strip() for status in deleted_files_statuses]
```

Comment:
it seems -1 element can not strip
status.split()[-1].strip()

Reply:
it will get the file name.